File: 0228-summary-ranges/0228-summary-ranges.kt
class Solution {
    fun summaryRanges(nums: IntArray): List<String> {
        val lst = ArrayList<String>()
        if (nums.size == 0) return lst
        var start = nums[0]
        for (i in 1..nums.lastIndex) {
            if (nums[i] != nums[i-1]+1) {
                if (start == nums[i-1]) {
                    lst.add("$start")
                } else {
                    lst.add("$start->${nums[i-1]}")
                }
                start = nums[i]
            }
        }
        if (start == nums[nums.lastIndex]) {
            lst.add("$start")
        } else {
            lst.add("$start->${nums[nums.lastIndex]}")
        }
        return lst
    }
}

File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        lst = []
        if len(nums) == 0:
            return lst
        start = nums[0]
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]+1:
                if start == nums[i-1]:
                    lst.append(str(start))
                else:
                    lst.append(str(start)+"->"+str(nums[i-1]))
                start = nums[i]
        if start == nums[-1]:
            lst.append(str(start))
        else:
            lst.append(str(start)+"->"+str(nums[-1]))
        return lst
